parity-clib/parity.h

```diff
@@ -14,166 +14,270 @@
+/// nullptr to disable logging to a file.  On Windows, this will be interpreted
+/// as UTF-8, not the system codepage, and is not limited to MAX_PATH.
+/// @param log_mode_len Length of the log_file or zero to disable logging to a
+/// file.
+/// @param logger Pointer to point to the created `Logger` object
 
-/// **Important**: This function must only be called exactly once otherwise it will panic. If you want to disable a
-/// logging mode or logging to a file make sure that you pass the `length` as zero
+/// **Important**: This function must only be called exactly once ―  otherwise,
+/// it will panic. If you want to disable a logging mode or logging to a file,
+/// make sure that you pass the `length` as zero
 ///
 /// # Example
 ///
 /// ```no_run
 /// void* cfg;
 /// const char *args[] = {"--light", "--can-restart"};
-/// size_t str_lens[] = {7, 13};
+/// uintptr_t str_lens[] = {7, 13};
 /// if (parity_config_from_cli(args, str_lens, 2, &cfg) != 0) {
-///		return 1;
+/// 		return 1;
 /// }
 /// char[] logger_mode = "rpc=trace";
-/// parity_set_logger(logger_mode, strlen(logger_mode), nullptr, 0, &cfg.logger);
+/// parity_set_logger(logger_mode, strlen(logger_mode), nullptr, 0,
+/// &cfg.logger);
 /// ```
 ///
-int parity_set_logger(const char* log_mode, size_t log_mode_len, const char* log_file, size_t log_file_len, void** logger);
+void parity_set_logger(const char *log_mode, uintptr_t log_mode_len,
+                       const char *log_file, uintptr_t log_file_len,
+                       struct parity_logger **logger);
 
 /// Destroys a configuration object created earlier.
 ///
-/// **Important**: You probably don't need to call this function. Calling `parity_start` destroys
-/// 				the configuration object as well (even on failure).
-void parity_config_destroy(void* cfg);
+/// **Important**: You probably don't need to call this function. Calling
+/// `parity_start` destroys the configuration object as well (even on failure).
+///
+/// It is safe to pass NULL here, in which case this function has no effect.
+void parity_config_destroy(struct parity_config *cfg);
 
-/// Starts the parity client in background threads. Returns a pointer to a struct that represents
-/// the running client. Can also return NULL if the execution completes instantly.
+/// Starts the parity client in a background thread.
 ///
-/// **Important**: The configuration object passed inside `cfg` is destroyed when you
-/// 				call `parity_start` (even on failure).
+/// **Important**: The configuration object passed inside `cfg` is destroyed
+/// when you call `parity_start` (even on failure).
 ///
-/// On success, the produced object will be written to the `void*` pointed by `out`.
+/// On success, the produced object will be written to the `void*` pointed by
+/// `out`.  If the provided command-line arguments didn’t require starting a
+/// node, `*out` will be unchanged.
 ///
 /// Returns 0 on success, and non-zero on error.
```

Comment:
I don't really understand why you kept the integer here, but not above.

Reply:
No good reason.

Reply:
Fixed with explicit error codes.